crates/bevy_text/src/pipeline.rs

```diff
@@ -57,159 +152,291 @@ impl TextPipeline {
-    ) -> Self {
-        let mut info = Self {
-            fonts: fonts.into_boxed_slice(),
-            sections: sections.into_boxed_slice(),
-            text_alignment,
-            linebreak_behavior,
-            min: Vec2::ZERO,
-            max: Vec2::ZERO,
-        };
+}
 
-        let min = info.compute_size(Vec2::new(0.0, f32::INFINITY));
-        let max = info.compute_size(Vec2::INFINITY);
-        info.min = min;
-        info.max = max;
-        info
+impl TextMeasureInfo {
+    pub fn compute_size(&self, bounds: Vec2) -> Vec2 {
+        let font_system = &mut self.font_system.try_lock().expect("Failed to acquire lock");
+        let mut buffer = self.buffer.lock().expect("Failed to acquire the lock");
+        buffer.set_size(font_system, bounds.x.ceil(), bounds.y.ceil());
+        buffer_dimensions(&buffer)
     }
+}
 
-    pub fn compute_size(&self, bounds: Vec2) -> Vec2 {
-        let sections = &self.sections;
-        let geom = SectionGeometry {
-            bounds: (bounds.x, bounds.y),
-            ..Default::default()
-        };
-        let section_glyphs = glyph_brush_layout::Layout::default()
-            .h_align(self.text_alignment.into())
-            .line_breaker(self.linebreak_behavior)
-            .calculate_glyphs(&self.fonts, &geom, sections);
-
-        compute_text_bounds(&section_glyphs, |index| {
-            let font = &self.fonts[index];
-            let font_size = self.sections[index].scale;
-            ab_glyph::Font::into_scaled(font, font_size)
-        })
-        .size()
+// /// get attr for from textstyle
+// /// loading fonts into the [`Database`](cosmic_text::fontdb::Database) if required.
+fn get_attrs<'a>(
+    section: &'a TextSection,
+    section_index: usize,
+    font_system: &mut cosmic_text::FontSystem,
+    map_handle_to_font_id: &mut HashMap<AssetId<Font>, cosmic_text::fontdb::ID>,
+    fonts: &Assets<Font>,
+) -> Attrs<'a> {
+    let font_handle = section.style.font.clone();
+    let font_handle_id = font_handle.id();
+    let face_id = map_handle_to_font_id
+        .entry(font_handle_id)
+        .or_insert_with(|| {
+            let font = fonts.get(font_handle).unwrap();
+            let data = Arc::clone(&font.data);
+            let ids = font_system
+                .db_mut()
+                .load_font_source(cosmic_text::fontdb::Source::Binary(data));
+            // TODO: it is assumed this is the right font face
+            *ids.last().unwrap()
+
+            // TODO: below may be required if we need to offset by the baseline (TBC)
+            // see https://github.com/pop-os/cosmic-text/issues/123
+            // let font = font_system.get_font(face_id).unwrap();
+            // map_font_id_to_metrics
+            //     .entry(face_id)
+            //     .or_insert_with(|| font.as_swash().metrics(&[]));
+        });
+    let face = font_system.db().face(*face_id).unwrap();
+
+    // TODO: validate this is the correct string to extract
```

Comment:
There's quite a lot of `TODO`s. Other than completing the ones necessary for this PR I think it would be better to aggregate the ones that are future work into an issue.